samplesheets/views_api.py

```diff
@@ -421,3 +421,40 @@ def get(self, request, **kwargs):
             except Exception as ex:
                 return Response(str(ex), status=500)
         return Response(ret, status=200)
+
+
+class ProjectIrodsFileListAPIView(SODARAPIBaseProjectMixin, APIView):
+    """
+    Return a list of files in the project's sample data repository.
+
+    **URL:** ``/samplesheets/api/irods/files/{Project.sodar_uuid}``
+
+    **Methods:** ``GET``
+
+    **Parameters:**
+
+    - ``project``: Project SODAR UUID
+
+    **Returns:**
+
+    - ``irods_data``: List of iRODS data objects
+    """
+
+    http_method_names = ['get']
+    permission_required = 'samplesheets.view_sheet'
+
+    def get(self, request, *args, **kwargs):
+        if not settings.ENABLE_IRODS:
+            raise APIException('iRODS not enabled')
+        irods_backend = get_backend_api('omics_irods')
+        project = self.get_project()
+        path = irods_backend.get_sample_path(project)
+        try:
+            with irods_backend.get_session() as irods:
+                irods_data = irods_backend.get_objects(irods, path)
+        except Exception as ex:
+            return Response(
+                {'detail': 'Exception queried iRODS objects: {}'.format(ex)},
```

Comment:
We should already be using a constant instead of a string for the error prefix here, and use that in tests. Also as a bit of proofreading, it could be improved into "Exception querying iRODS objects".